src/scenic/simulators/newtonian/simulator.py

```diff
@@ -196,7 +196,22 @@ def isOnScreen(self, x, y):
     def step(self):
         for obj in self.objects:
             current_speed = obj.velocity.norm()
-            if hasattr(obj, "hand_brake"):
+            # 1) Pedestrian: we stash .control['heading']/['speed']
+            if hasattr(obj, "control") and "speed" in obj.control:
+                h = (
+                    obj.control["heading"]
+                    if obj.control["heading"] is not None
+                    else obj.heading
+                )
+                s = (
+                    obj.control["speed"]
+                    if obj.control["speed"] is not None
```

Comment:
If the `heading` and `speed` controls are both `None`, this will change the pedestrian to be moving in the direction `obj.heading`, even if it wasn't before (e.g. if you manually set the velocity so the pedestrian is moving in some other direction: `new Pedestrian with velocity (1, 1)`). If the pedestrian doesn't take any walking actions, we should leave its velocity unchanged.